autoload/tagbar/types/uctags.vim

```diff
@@ -957,6 +957,35 @@ function! tagbar#types#uctags#init(supported_types) abort
         \ {'short' : 'p', 'long' : 'procedures', 'fold' : 0, 'stl' : 1}
     \ ]
     let types.tcl = type_tcl
+    " TypeScript {{{1
+    let type_ts = tagbar#prototypes#typeinfo#new()
+    let type_ts.ctagstype = 'typescript'
+    let type_ts.kinds = [
+        \ {'short' : 'n', 'long' : 'namespaces',    'fold' : 0, 'stl' : 1},
+        \ {'short' : 'i', 'long' : 'interfaces',    'fold' : 0, 'stl' : 1},
+        \ {'short' : 'g', 'long' : 'enums',         'fold' : 0, 'stl' : 1},
+        \ {'short' : 'e', 'long' : 'enumerators',   'fold' : 0, 'stl' : 1},
```

Comment:
Does it need both `enums` and `enumerators`? Does ctags output two different types?

Reply:
Ah, that should read `enumerations` -- the actual members of an enum. From the sample file above:

```
enum Test {
    FOO
}

Test    test.ts      /^enum Test {$/;"       g
FOO     test.ts      /^  FOO,$/;"            e       enum:Test
```